deploy/datadog-agent/etc/datadog-agent/checks.d/json_check.py

```diff
@@ -20,28 +22,29 @@ def check(self, instance):
         # Make a get request
         try:
             r = requests.get(url, timeout=default_timeout)
-
-            if r.status_code == 200:
-                data = json.loads(r.text)
-                for metric in metrics:
-                    for name, path in metric.items():
-                        try:
-                            # Get the value from the path in the json file
-                            expression = parse(path)
-                            match = expression.find(data)[0].value
-                            self.gauge(name, float(match))
-                        except IndexError:
-                            # If there is no matching value
-                            self.no_match_event(url, path, aggregation_key)
-                        except ValueError:
-                            # If the matching value is not a number
-                            self.unexpected_match_event(url, path, match, aggregation_key)
-            else:
-                # If the request was not successful
-                self.status_code_event(url, r, aggregation_key)
         except requests.exceptions.Timeout:
-            # If there's a timeout
             self.timeout_event(url, default_timeout, aggregation_key)
+            return
+
+        if not r.ok:
+            # If the request was not successful
+            self.status_code_event(url, r, aggregation_key)
```

Comment:
if you only use the status code in this function (according to function name), pass just `r.status_code`

Reply:
the exception handling here is problematic. What if you get a `ConnectionException`? It will fail without proper exception handling. Consider using the standard approach outlined in https://stackoverflow.com/questions/16511337/correct-way-to-try-except-using-python-requests-module:
```python
try:
    r = requests.get(url,timeout=3)
    r.raise_for_status()
except requests.exceptions.HTTPError as errh:
    logger.log("Http Error:",errh)
    return
except requests.exceptions.ConnectionError as errc:
    logger.log("Error Connecting:",errc)
    return
except requests.exceptions.Timeout as errt:
    logger.log("Timeout Error:",errt)
    return
except requests.exceptions.RequestException as err:
    logger.log("OOps: Something Else",err)
    return
```